### PR TITLE
Use remote ampstate

### LINKED ISSUE #832: sample: data fetch for amp-state on page load (i.e. amp-state[src])
demo's ability to have ultra-fresh data after user interaction

### PR DESCRIPTION
**Pull request by kul3r4**
Fixes #832.
@ericlindley-g @choumx I am not sure about the scenario where the src URL of amp-state is updated and how to setup the src default value. The only way I found was to specify it inside the json itself but it seems weird. If I don't specify a default, then the validator complains.

### CODE REVIEW

**Comment by choumx** [comment]
Not sure I understand what this example is demonstrating right now.

Another way to look at the feature is that it allows fetching more `amp-state` data dynamically (instead of having everything at page load).

The I/O codelab does this to fetch additional product data, a common e-commerce pattern: https://codelabs.developers.google.com/codelabs/advanced-interactivity-in-amp/#0

I think we can similarly extend the ABE amp-bind example by adding a third animal (bird?) that's fetched dynamically via `[src]` binding. E.g.

```html
<amp-state id="allAnimals" [src]="remoteAnimalUrl">...</amp-state>

<button on="tap:AMP.setState({remoteAnimalUrl: '...'})>Set to bird</button>
```

**Comment by kul3r4** [comment]
@choumx thanks for explaining, it makes more sense now! I updated the sample accordingly.
@sebastianbenz Could you review it?

File: src/20_Components/amp-bind.html [hunk 1]
```diff
@@ -158,7 +165,19 @@
...
+    </button>
+
+    <!-- ## Load state remotely -->
+    <!--
+      By setting the src value of the `amp-state`, new data are fetched remotely and merged into the `amp-state` data json.
+    -->
+
+    <button class="ampstart-btn caps m1"
+            on="tap:AMP.setState({remoteAnimalUrl: '/json/amp-bind-data-new-state.json?RANDOM'})">
+                                  Set to Bird
```
Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: formatting looks a bit off here

File: src/20_Components/amp-bind.html [hunk 2]
```diff
@@ -158,7 +165,19 @@
...
     <button class="ampstart-btn caps m1"
-            on="tap:AMP.setState({currentAnimal: 'cat'})">Set to Cat</button>
-  </section>
+            on="tap:AMP.setState({allAnimals: {currentAnimal: 'cat'}})">
+                                  Set to Cat
+    </button>
+
+    <!-- ## Load state remotely -->
+    <!--
+      By setting the src value of the `amp-state`, new data are fetched remotely and merged into the `amp-state` data json.
```
Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: s/src value/`src` attribute/ (including the code formatting for src)
s/data are/data is/ 
remove 'remotely' and 'data json'  (not needed)

Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: Does it make sense to show the new json data inline?

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

File: src/20_Components/amp-bind.html [hunk 3]
```diff
@@ -108,16 +111,16 @@
     <!--
       The text of this `<p>` will change when the `currentAnimal` variable
       changes. `currentAnimal` is an uninitialized implicit state variable, which will be
-      set when a state change is triggered (see below).
+      set when a state change is triggered (see below). A state variable default can be specified using amp-state.
```
Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: can you please mark amp-state as code

File: src/20_Components/amp-bind.html [hunk 4]
```diff
@@ -157,8 +161,65 @@
...
+    <button class="ampstart-btn caps m1" on="tap:AMP.setState({allAnimals: {currentAnimal: 'cat'}})">
+      Set to Cat
+    </button>
+
+    <!-- ## Configuring initial state remotely -->
+    <!--
+      You can also bind to the `amp-state` `src` to provide data remotely.
+      The remote data returned from this fetch is merged into the `amp-state` json data, for example:
+      ```json
+      { "currentAnimal": "bird",
```
Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: nit: JSON formatting, add a line break before "currentAnimal"

Comment (sebastianbenz) [inline src/20_Components/amp-bind.html]: ping

**Comment by kul3r4** [comment]
@sebastianbenz somehow I missed that line, I added it now

**Pull request merged**